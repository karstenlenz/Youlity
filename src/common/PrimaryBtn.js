import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components/macro'

PrimaryBtn.propTypes = {
  onClick: PropTypes.func,
  children: PropTypes.string.isRequired,
}

export default function PrimaryBtn({ children, onClick }) {
  return <PrimaryBtnStyled onClick={onClick}>{children}</PrimaryBtnStyled>
}

const PrimaryBtnStyled = styled.button`
  font-size: 1.21rem;
  text-decoration: none;
  font-family: 'Ubuntu', sans-serif;
  color: white;
  height: 50px;
  display: block;
  padding: 10px;
  width: 100%;
  background: var(--primary);
  border: none;
  border-radius: 39px;

  &:disabled {
    opacity: 0.3;
    color:color: rgba(0, 0, 0, 0.5) ;
  }
`
