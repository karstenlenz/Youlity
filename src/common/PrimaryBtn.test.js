import React from 'react'
import { render, screen } from '@testing-library/react'
import PrimaryBtn from './PrimaryBtn'

describe('The PrimaryBtn component', () => {
  it('renders correctly', () => {
    const { container } = render(<PrimaryBtn>Test-Text</PrimaryBtn>)
    expect(container).toMatchSnapshot()
  })

  it('display the text "Test-Text"', () => {
    render(<PrimaryBtn>Test-Text</PrimaryBtn>)
    expect(screen.getByText('Test-Text')).toBeInTheDocument()
  })
})
