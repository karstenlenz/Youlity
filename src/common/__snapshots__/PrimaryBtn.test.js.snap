// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The PrimaryBtn component renders correctly 1`] = `
<div>
  <button
    class="PrimaryBtn__PrimaryBtnStyled-sc-70yx6k-0 cswOOA"
  >
    Test-Text
  </button>
</div>
`;
